### PR TITLE
EP-215 PBBA integration

### PR DESCRIPTION
**Pull request by AlexeiJovmir**
added PBBA type
added service and configs for PBBA
logic for PBBA, handle error in response.
PBBA switch in settings
Important: there is no call back from bank app!

### CODE REVIEW

File: Examples/ObjectiveCExampleApp/ObjectiveCExampleApp.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -678,15 +678,16 @@
...
 				CURRENT_PROJECT_VERSION = 5;
-				DEVELOPMENT_TEAM = "";
+				DEVELOPMENT_TEAM = F778669843;
 				INFOPLIST_FILE = ObjectiveCExampleApp/Info.plist;
 				LD_RUNPATH_SEARCH_PATHS = (
 					"$(inherited)",
 					"@executable_path/Frameworks",
 				);
-				PRODUCT_BUNDLE_IDENTIFIER = com.judo.JudoPayDemoObjC;
+				PRODUCT_BUNDLE_IDENTIFIER = com.judo.JudoPayDemoObjC1;
```
Comment (mpetrenco) [inline Examples/ObjectiveCExampleApp/ObjectiveCExampleApp.xcodeproj/project.pbxproj:690]: ?

File: Examples/ObjectiveCExampleApp/Podfile.lock [hunk 2]
```diff
@@ -30,10 +33,11 @@ SPEC CHECKSUMS:
...
   InAppSettingsKit: 4890a44f9df7a1742c632920f0ea939cbeb29ea2
-  JudoKitObjC: fc659a2d67db3ef22b18ef4014fb9005531bc048
+  JudoKitObjC: 799a126da822eef6ec2025676202da5f5d9e2918
   PayCardsRecognizer: 2e4d1b2fdd5f1528fb56064a91e1d58a1af7a2db
   TrustKit: 073855e3adecd317417bda4ac9e9ac54a2e3b9f2
+  ZappMerchantLib: b14bc5814840426d351190309250347ca9b0983d
 
 PODFILE CHECKSUM: e20a57c9c0cbf8a9f4490d8e0ae390f4f56fd514
 
-COCOAPODS: 1.9.1
```
Comment (mpetrenco) [inline Examples/ObjectiveCExampleApp/Podfile.lock:39]: update your cocoapods version please

File: Source/Extensions/NSError/NSError+Additions.h [hunk 3]
```diff
@@ -58,6 +58,7 @@ extern NSString *_Nonnull const JudoErrorDomain;
 + (nonnull NSError *)judoInvalidIDEALCurrencyError;
 + (nonnull NSError *)judoApplePayNotSupportedError;
 + (nonnull NSError *)judoSiteIDMissingError;
++ (nonnull NSError *)judoInvalidPBBACurrencyPound;
```
Comment (mpetrenco) [inline Source/Extensions/NSError/NSError+Additions.h:61]: remove Pound at the end please

File: Source/Modules/PaymentMethods/Interactor/JPPaymentMethodsInteractor.m [hunk 4]
```diff
@@ -149,6 +151,12 @@ - (JPAmount *)getAmount {
     } else {
         [self removePaymentMethodWithType:JPPaymentMethodTypeIDeal];
     }
+    
+    if ([self.configuration.amount.currency isEqualToString:kCurrencyPounds]) { // &&  ([PBBAAppUtils isCFIAppAvailable])) {
```
Comment (mpetrenco) [inline Source/Modules/PaymentMethods/Interactor/JPPaymentMethodsInteractor.m:155]: Do we need the comment ?

File: Source/Modules/PaymentMethods/Interactor/JPPaymentMethodsInteractor.m [hunk 5]
```diff
@@ -222,6 +236,15 @@ - (JPApplePayService *)applePayService {
     return _applePayService;
 }
 
+
+- (JPPBBAService *)pbbaService {
+    if (!_pbbaService && self.configuration) {
+        _pbbaService = [[JPPBBAService alloc] initWithConfiguration:self.configuration
+                                                         transactionService:self.transactionService];
```
Comment (mpetrenco) [inline Source/Modules/PaymentMethods/Interactor/JPPaymentMethodsInteractor.m:243]: clang-format.sh please

File: Source/Services/PBBA/JPPBBAService.h [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+#import <Foundation/Foundation.h>
+#import <ZappMerchantLib/PBBAAppUtils.h>
+
+@interface JPPBBAService : NSObject
+
+/**
+ * Creates an instance of an JPPBBAService object
+ *
+ * @param configuration - an instance of JPConfiguration used to configure the PBBA
+ * @param transactionService - an instance of JPTransactionService responsible for Judo backend calls
```
Comment (mpetrenco) [inline Source/Services/PBBA/JPPBBAService.h:36]: Documentations must also specify the return type

File: Source/Services/PBBA/JPPBBAService.m [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+
+- (instancetype)initWithConfiguration:(JPConfiguration *)configuration
+                   transactionService:(JPTransactionService *)transactionService {
+    if (self = [super init]) {
+        self.configuration = configuration;
+        self.transactionService = transactionService;
+    }
+    
+    [UIApplication.topMostViewController.view addSubview:self.transactionStatusView];
+    [self.transactionStatusView pinToView:UIApplication.topMostViewController.view withPadding:0.0];
```
Comment (mpetrenco) [inline Source/Services/PBBA/JPPBBAService.m:64]: Services handling business logic should not handle UI stuff as well.

File: Source/Services/PBBA/JPPBBAService.m [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+        @"amount" : @(amount.amount.doubleValue),
+        @"country" : @"GB",
+        @"accountHolderName" : @"PBBA User",
+        @"merchantPaymentReference" : reference.paymentReference,
+        @"bic" : @"RABONL2U",
+        @"merchantConsumerReference" : reference.consumerReference,
+        @"siteId" : self.configuration.siteId
+    }];
+    
+    if (self.configuration.pbbaConfiguration.mobileNumber) [parameters setValue:self.configuration.pbbaConfiguration.mobileNumber forKey:@"mobileNumber"];
```
Comment (mpetrenco) [inline Source/Services/PBBA/JPPBBAService.m:196]: would've preferred using {} for long if statements

File: Source/Services/PBBA/JPPBBAService.m [hunk 9]
```diff
@@ -0,0 +1,210 @@
...
+        if (data.orderDetails.orderId && data.redirectUrl) {
+            [weakSelf remapPBBAResponse:response completion: completion];
+            return;
+        }
+        
+        completion(nil, NSError.judoResponseParseError);
+    }];
+}
+
+- (void)remapPBBAResponse:(JPResponse *)response completion:(JudoCompletionBlock)completion  {
```
Comment (mpetrenco) [inline Source/Services/PBBA/JPPBBAService.m:102]: The method displays the PBBA popup and handles polling. Method name does not match functionality.

File: Source/Services/PBBA/JPPBBAService.h (hunk 6 above)
Comment (AlexeiJovmir) [reply Source/Services/PBBA/JPPBBAService.h:36]: * Creates an instance of an JPPBBAService object

File: Examples/ObjectiveCExampleApp/ObjectiveCExampleApp.xcodeproj/project.pbxproj [hunk 10]
```diff
@@ -678,9 +678,10 @@
 			buildSettings = {
 				ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES = YES;
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
-				CODE_SIGN_STYLE = Manual;
+				CODE_SIGN_IDENTITY = "Apple Development";
+				CODE_SIGN_STYLE = Automatic;
 				CURRENT_PROJECT_VERSION = 5;
-				DEVELOPMENT_TEAM = "";
+				DEVELOPMENT_TEAM = F778669843;
```
Comment (stpopa) [inline Examples/ObjectiveCExampleApp/ObjectiveCExampleApp.xcodeproj/project.pbxproj:684]: please revert code signing changes

File: JudoKitObjC.xcodeproj/project.pbxproj [hunk 11]
```diff
@@ -2228,6 +2311,7 @@
 			buildActionMask = 2147483647;
 			files = (
 				30D0894023901F1F00480C9E /* icons.bundle in Resources */,
+				0701434824642AF100C5D299 /* ZappMerchantLib.framework.dSYM in Resources */,
```
Comment (stpopa) [inline JudoKitObjC.xcodeproj/project.pbxproj:2314]: why we need a reference of `ZappMerchantLib.framework.dSYM` here?

**Comment by stpopa** [comment]
Covered by : https://github.com/Judopay/JudoKit-iOS/pull/15

**Pull request closed without merging**